posthog/hogql/functions/mapping.py

```diff
@@ -69,21 +107,189 @@ class HogQLFunctionMeta:
+            ),
+            ((DateTimeType(), IntegerType()), DateTimeType()),
+            ((IntegerType(), DateTimeType()), DateTimeType()),
+        ],
+    ),
+    "intDiv": HogQLFunctionMeta(
+        "intDiv",
+        2,
+        2,
+        signatures=[
+            ((IntegerType(), IntegerType()), IntegerType()),
+        ],
+    ),
+    "intDivOrZero": HogQLFunctionMeta(
+        "intDivOrZero",
+        2,
+        2,
+        signatures=[
+            ((IntegerType(), IntegerType()), IntegerType()),
+        ],
+    ),
+    "modulo": HogQLFunctionMeta(
+        "modulo",
+        2,
+        2,
+        signatures=[
+            ((IntegerType(), IntegerType()), IntegerType()),
+            ((FloatType(), IntegerType()), FloatType()),
+            ((IntegerType(), FloatType()), FloatType()),
+        ],
+    ),
+    "moduloOrZero": HogQLFunctionMeta(
+        "moduloOrZero",
+        2,
+        2,
+        signatures=[
+            ((IntegerType(), IntegerType()), IntegerType()),
+            ((FloatType(), IntegerType()), FloatType()),
+            ((IntegerType(), FloatType()), FloatType()),
+        ],
+    ),
+    "positiveModulo": HogQLFunctionMeta(
+        "positiveModulo",
+        2,
+        2,
+        signatures=[
+            ((IntegerType(), IntegerType()), IntegerType()),
+            ((FloatType(), IntegerType()), FloatType()),
+            ((IntegerType(), FloatType()), FloatType()),
+        ],
+    ),
+    "negate": HogQLFunctionMeta(
+        "negate",
+        1,
+        1,
+        signatures=[
+            ((IntegerType(),), IntegerType()),
+            ((FloatType(),), FloatType()),
+        ],
+    ),
+    "abs": HogQLFunctionMeta(
+        "abs",
+        1,
+        1,
+        signatures=[
+            ((IntegerType(),), IntegerType()),
+        ],
+        case_sensitive=False,
+    ),
+    "gcd": HogQLFunctionMeta(
+        "gcd",
+        2,
+        2,
+        signatures=[
+            ((IntegerType(),), IntegerType()),
+        ],
+    ),
+    "lcm": HogQLFunctionMeta(
+        "lcm",
+        2,
+        2,
+        signatures=[
+            ((IntegerType(),), IntegerType()),
+        ],
+    ),
+    "max2": HogQLFunctionMeta(
+        "max2",
+        2,
+        2,
+        signatures=[
+            ((IntegerType(), IntegerType()), FloatType()),
+            ((FloatType(), IntegerType()), FloatType()),
+            ((IntegerType(), FloatType()), FloatType()),
+        ],
+        case_sensitive=False,
+    ),
+    "min2": HogQLFunctionMeta(
+        "min2",
+        2,
+        2,
+        signatures=[
+            ((IntegerType(), IntegerType()), FloatType()),
+            ((FloatType(), IntegerType()), FloatType()),
+            ((IntegerType(), FloatType()), FloatType()),
+        ],
+        case_sensitive=False,
+    ),
     "multiplyDecimal": HogQLFunctionMeta("multiplyDecimal", 2, 3),
     "divideDecimal": HogQLFunctionMeta("divideDecimal", 2, 3),
     # arrays and strings common
```

Comment:
NIT: add comment here that these signatures are incomplete/what's required to add them—just put in the signature or have to add a test for each one etc